### Code review

Found 2 issues:

1. **`confirm_or_exit` 不检查 `safe_confirm` 返回值——TTY 用户拒绝时危险操作继续执行**

   `safe_confirm` 在 TTY 分支走 `bool(typer.confirm(message, default=default))`，而 `typer.confirm` 默认 `abort=False`，用户输 `n` 时**返回 False 而非 raise**（click 源码 + 实测确认）。`confirm_or_exit` 调用 `safe_confirm(...)` 后**完全忽略返回值**（cli_utils.py:289），用户拒绝时函数正常返回 None，命令继续执行 destructive 操作。

   docstring 声称"TTY：走 typer.confirm，用户拒绝则 typer 自己 Exit"——这是错误前提，`typer.confirm` 不带 `abort=True` 时不会自己 Exit。

   **触发场景**：TTY 下跑 `ginkgo param delete -p <uuid>`（或 `cache clear` / `kafka reset` / `kafka purge`）不传 `--force`，提示出现后输 `n`。**后果**：用户以为已取消，但 `delete_by_uuid` / `_clear_all_caches` / `kafka_topic_set` / 清队列消息照常执行 → 数据丢失。master 原代码用 `if not Confirm.ask(...): return / raise typer.Exit(0)` 显式检查返回值，本 PR 迁移后这层守卫整个丢了，是 regression。

   **实测**：mock `stdin.isatty=True` + `typer.confirm` 返 False → `confirm_or_exit` 返回 None 无异常（`/home/kaoru/.ginkgo/.venv/bin/python` 跑 PR head 的 `cli_utils.py`）。

   **测试盲区**：4 个 `test_confirmed_proceeds_with_*` 全 mock `safe_confirm=True`，无 False 分支，所以漏了。建议补一条 TTY-reject 单测。

   修复方向（任一）：`confirm_or_exit` 检查返回值 False 时 `raise typer.Exit(0/1)`；或 `safe_confirm` TTY 分支用 `typer.confirm(..., abort=True)` 让拒绝抛 Abort 后 catch 转 Exit。

   <https://github.com/Kaoruha/GinkGO/blob/8199edc10fbc066c556297b1f83984e6db833cd8/src/ginkgo/client/cli_utils.py#L275-L294>

2. **`kafka purge` flag `--yes/-y/--confirm` → `--force/-f` 破坏既有测试 + 违反 #6006/#6390 统一方向**

   (a) **破坏既有测试**：master 上 `tests/unit/client/test_cli_confirm_flag.py:34` 参数化矩阵含 `("ginkgo.client.kafka_cli", "app", ["purge"])`，断言 `purge --help` 必须含 `--yes` 和 `-y`（#6390 锁定）。本 PR 把 purge flag 改 `--force/-f` 但未更新该测试。**实测 PR head 跑该测试：1 failed**（`AssertionError: ginkgo.client.kafka_cli ['purge']: --yes 未注册`）。PR body 声称"Layer 3 变更相关 53 passed"是漏跑 `test_cli_confirm_flag.py` 的假绿——实际 5 个相关测试文件合跑 58 passed / **1 failed**。

   (b) **违反 maintainer 在 #6006/#6390 明确的统一方向**：「统一为 `--yes/-y`（旧 `--confirm` 保留别名），`--force` 语义不同不含」。#6390 正是把 purge 从 `--confirm` 统一到 `--yes/-y/--confirm` 的 PR。master 4 命令里 `kafka purge` 是**唯一**对齐 #6006 的，本 PR 把它拉回 `--force`，#6006 AC1 在该子区域覆盖率 1/4 → 0/4。

   PR body 辩护"与其他 3 命令统一"，但 #6006/#6390 方向是反向（全 → `--yes/-y`），不是全 → `--force`。正确做法是把另 3 个命令的 `--force/-f` 也改 `--yes/-y/--confirm`。

   <https://github.com/Kaoruha/GinkGO/blob/8199edc10fbc066c556297b1f83984e6db833cd8/src/ginkgo/client/kafka_cli.py#L43-L45>

🤖 Generated with [Claude Code](https://claude.ai/code)

<sub>- If this code review was useful, please react with 👍. Otherwise, react with 👎.</sub>

--- a/src/ginkgo/client/cli_utils.py
+++ b/src/ginkgo/client/cli_utils.py
@@ -275,22 +275,32 @@ def safe_confirm(
 def confirm_or_exit(message: str, *, yes_flag: bool = False) -> None:
     """safe_confirm 命令层封装（ADR-021 第 3 维，#6578）。
 
-    危险操作命令的统一确认入口：非 TTY 且未提供 ``--force`` 时
-    ``safe_confirm`` 抛 ``CliConfirmError``，此处按具体类型 catch
-    （不被通用 ``except Exception`` 吞，呼应 ``CliConfirmError`` 设计）
-    → stderr 输出错误 + ``typer.Exit(1)``。非零退出码让 CI/脚本能
-    检测到"操作被拒绝"，而非静默 no-op 误以为成功。
+    危险操作命令的统一确认入口：
+    - ``yes_flag=True``（命令的 ``--force``/``--yes``）：safe_confirm 短路返 True，放行
+    - TTY + 用户确认（typer.confirm 返 True）：放行
+    - TTY + 用户拒绝（typer.confirm 返 False）：``typer.Exit(0)`` —— 用户主动
+      取消，正常退出（对齐 master 原契约 ``if not Confirm.ask(...): return/Exit(0)``）
+    - 非 TTY + 无 yes_flag：safe_confirm 抛 ``CliConfirmError`` → stderr + ``typer.Exit(1)``。
+      非零退出码让 CI/脚本能检测到"操作被拒绝"，而非静默 no-op 误以为成功。
 
-    - ``yes_flag=True``（命令的 ``--force``）：safe_confirm 短路返 True，放行
-    - TTY：走 typer.confirm，用户拒绝则 typer 自己 Exit
-    - 非 TTY + 无 yes_flag：本函数 Exit(1)
+    ``CliConfirmError`` 按具体类型 catch（不被通用 ``except Exception`` 吞，
+    呼应其设计），错误走 stderr 保持 stdout（脚本数据通道）干净。
+
+    .. note::
+       ``typer.confirm`` 默认 ``abort=False``，拒绝返 False **不** raise，
+       故必须显式检查 ``safe_confirm`` 返回值，不能依赖 typer 自己 Exit
+       （#6578 review #1 的 regression：原实现忽略返回值，用户拒绝仍执行
+       destructive op，把 master 守卫整个删了）。
     """
     try:
-        safe_confirm(message, yes_flag=yes_flag)
+        confirmed = safe_confirm(message, yes_flag=yes_flag)
     except CliConfirmError as e:
         # 错误走 stderr：保持 stdout（脚本数据通道）干净
         Console(stderr=True).print(f"[red]Error:[/red] {e}")
         raise typer.Exit(1)
+    if not confirmed:
+        # TTY 用户主动拒绝（safe_confirm 走 typer.confirm 返 False）
+        raise typer.Exit(0)
 
 
 def make_progress(*, format: str, isatty: bool) -> Optional["Progress"]:

--- a/src/ginkgo/client/kafka_cli.py
+++ b/src/ginkgo/client/kafka_cli.py
@@ -41,11 +41,11 @@ def reset(
 @app.command()
 def purge(
     queue_name: str = typer.Argument(..., help="要清理的队列名称"),
-    force: bool = typer.Option(False, "--force", "-f", help="跳过确认提示")
+    confirm: bool = typer.Option(False, "--yes", "-y", "--confirm", help="跳过确认提示"),
 ):
     """清理指定队列的所有消息"""
     console.print(f"[bold red][red]:wastebasket:[/red] Purging queue: {queue_name}[/]")
-    _purge_queue_messages(queue_name, force)
+    _purge_queue_messages(queue_name, confirm)
 
 
 @app.command()
@@ -231,7 +231,7 @@ def _reset_kafka_queues(queue_name: Optional[str], force: bool):
         console.print(f"[red]{traceback.format_exc()}[/]")
 
 
-def _purge_queue_messages(queue_name: str, force: bool):
+def _purge_queue_messages(queue_name: str, confirm: bool):
     """清理队列消息 - 使用KafkaService实现"""
     try:
         from ginkgo.data.containers import container
@@ -239,7 +239,7 @@ def _purge_queue_messages(queue_name: str, force: bool):
 
         console.print(f"[yellow]:warning: You are about to purge ALL messages from queue: '{queue_name}'[/]")
         console.print("[red]This will permanently delete all pending messages![/]")
-        confirm_or_exit("[bold red]Are you sure you want to continue?[/]", yes_flag=force)
+        confirm_or_exit("[bold red]Are you sure you want to continue?[/]", yes_flag=confirm)
 
         console.print(f"[red]:wastebasket: Purging messages from queue: {queue_name}[/]")
 

--- a/tests/unit/client/test_cli_utils_confirm.py
+++ b/tests/unit/client/test_cli_utils_confirm.py
@@ -35,3 +35,18 @@ class TestConfirmOrExit:
             with patch("ginkgo.client.cli_utils.typer.confirm", return_value=True) as m:
                 confirm_or_exit("dangerous op")
             m.assert_called_once()
+
+    def test_tty_user_rejects_raises_exit0(self):
+        """TTY + 用户拒绝（typer.confirm 返 False）→ typer.Exit(0)。
+
+        master 原契约：用户主动取消 = 正常退出（exit 0），且不能继续
+        执行危险操作。typer.confirm 默认 abort=False，拒绝返 False 不
+        raise，故 confirm_or_exit 必须显式检查 safe_confirm 返回值
+        （#6578 review #1 抓到的 regression：原实现忽略返回值，用户拒绝
+        仍执行 destructive op，等于把 master 守卫整个删了）。
+        """
+        with patch("sys.stdin.isatty", return_value=True):
+            with patch("ginkgo.client.cli_utils.typer.confirm", return_value=False):
+                with pytest.raises(typer.Exit) as exc_info:
+                    confirm_or_exit("dangerous op")
+        assert exc_info.value.exit_code == 0

--- a/tests/unit/client/test_kafka_cli.py
+++ b/tests/unit/client/test_kafka_cli.py
@@ -124,7 +124,7 @@ class TestReset:
 class TestPurge:
     """Tests for the 'purge' command."""
 
-    def test_purge_with_force(self, cli_runner):
+    def test_purge_with_confirm(self, cli_runner):
         mock_service = MagicMock()
         mock_service.topic_exists.return_value = True
         mock_service.get_message_count.return_value = 10
@@ -134,7 +134,7 @@ class TestPurge:
 
         with patch("ginkgo.data.containers.container") as mock_container:
             mock_container.kafka_service.return_value = mock_service
-            result = cli_runner.invoke(kafka_cli.app, ["purge", "test_topic", "--force"])
+            result = cli_runner.invoke(kafka_cli.app, ["purge", "test_topic", "--yes"])
 
         assert result.exit_code == 0
 
@@ -176,11 +176,11 @@ class TestResetConfirm:
 class TestPurgeConfirm:
     """kafka purge TTY 守卫（ADR-021 E3, #6578）。
 
-    场景 2（非 TTY + --force 执行）已由 TestPurge::test_purge_with_force 覆盖。
+    场景 2（非 TTY + --yes 执行）已由 TestPurge::test_purge_with_confirm 覆盖。
     """
 
-    def test_non_tty_without_force_exits1(self, cli_runner):
-        """非 TTY 无 --force → Exit(1)，kafka_service 未调用。"""
+    def test_non_tty_without_yes_exits1(self, cli_runner):
+        """非 TTY 无 --yes → Exit(1)，kafka_service 未调用。"""
         with patch("ginkgo.data.containers.container") as mock_container:
             mock_container.kafka_service.return_value = MagicMock()
             result = cli_runner.invoke(kafka_cli.app, ["purge", "test_topic"])
@@ -442,7 +442,7 @@ class TestKafkaCLIExceptions:
 
         with patch("ginkgo.data.containers.container") as mock_container:
             mock_container.kafka_service.return_value = mock_service
-            result = cli_runner.invoke(kafka_cli.app, ["purge", "nonexistent_topic", "--force"])
+            result = cli_runner.invoke(kafka_cli.app, ["purge", "nonexistent_topic", "--yes"])
 
         assert result.exit_code == 0
         assert "does not exist" in result.output.lower()

